python/ray/data/read_api.py

```diff
@@ -600,6 +610,16 @@ def from_pandas_refs(dfs: Union[ObjectRef["pandas.DataFrame"], List[ObjectRef[
     """
     if isinstance(dfs, ray.ObjectRef):
         dfs = [dfs]
+    else:
+        for df in dfs:
+            assert isinstance(df, ray.ObjectRef), df
```

Comment:
Should raise an exception instead of making an assertion for user-facing input validation:
```suggestion
    if isinstance(dfs, ray.ObjectRef):
        dfs = [dfs]
    elif isinstance(dfs, list):
        for df in dfs:
            if not isinstance(df, ray.ObjectRef):
                raise ValueError(
                    f"Expected list of Ray object refs, got list containing {type(df)}")
    else:
        raise ValueError(
            f"Expected Ray object ref or list of Ray object refs, got {type(df)}")
```